examples/e19_instant_repl.clj

```diff
@@ -31,25 +34,37 @@
                               :on-text-changed {:event/type ::text-changed}}
                              {:fx/type :scroll-pane
                               :grid-pane/column 1
-                              :content {:fx/type :label
-                                        :style {:-fx-font-family "monospace"}
-                                        :wrap-text true
-                                        :text (with-out-str
-                                                (try
-                                                  (let [form (read-string text)]
-                                                    (pprint/pprint form)
-                                                    (print "=> ")
-                                                    (pprint/pprint (eval form)))
-                                                  (catch Exception e
-                                                    (pprint/pprint e))))}}]}}})
+
+                              :content
+                              {:fx/type :label
+                               :style {:-fx-font-family "monospace"}
+                               :wrap-text true
+                               :text
+                               (with-out-str
+                                 (let [EOF (Object.)
+                                       rdr
+                                       (-> text char-array (io/reader)
+                                           BufferedReader.
+                                           LineNumberingPushbackReader.)]
+                                   (try (loop []
+                                          (let [form (read {:eof EOF} rdr)]
+                                            (when-not (identical? form EOF)
+                                              (pprint/pprint form)
+                                              (print ";=> ")
+                                              (pprint/pprint (eval form))
+                                              (recur))))
+                                        (catch Throwable ex
+                                          (pprint/pprint
+                                            (ex-data ex))))))}}]}}})
```

Comment:
Typing `(range "")` in input field should report an error, but instead it shows `nil` because raised exception has no ex-data. I'd prefer to have this here:
```clj
(println
  (-> ex Throwable->map clojure.main/ex-triage clojure.main/ex-str))
```

Reply:
Updated